trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1394,7 +1394,16 @@ public Response deleteDerivedProvider(
             if (configurator == null) {
                 return Response.status(Status.NOT_FOUND).entity(NO_SUCH_PROVIDER).build();
             }
+
+            // Get all DP descriptors before they are removed
+            List<IDataProviderDescriptor> allDerivedDps = manager.getAvailableProviders(experiment, derivedDescriptor.getConfiguration());
+
+            // Clean-up configuration
             configurator.removeDataProviderDescriptor(experiment, derivedDescriptor);
+
+            // Remove any derived data provider instances stored in the manager
+            allDerivedDps.forEach(desc -> manager.removeDataProvider(experiment, desc.getId()));
```

Comment:
I found that it seems to work if you remove the data providers first and then remove the derived descriptor and configuration. Is there a reason to remove them after?

Reply:
No there is no particular reason. I can call it first as well.